test/framework/test_framework.py

```diff
@@ -340,3 +340,32 @@ def circuits(self):
         }
 
         self.assertEqual(exp.job_info(backend=backend), job_info)
+
+    @ddt.data((None, "Test1"), (None, "Test2"))
+    @ddt.unpack
+    def test_experiment_type(self, exp_type1, exp_type2):
+        """Test the experiment_type setter for the experiment."""
+
+        class MyExp(BaseExperiment):
+            """Some arbitrary experiment"""
+
+            def __init__(self, *args, **kwargs):
+                super().__init__(*args, **kwargs)
+
+            def circuits(self):
+                pass
+
+        exp1 = MyExp(physical_qubits=[0], experiment_type=exp_type1)
+        if exp_type1 is None:
+            exp_type1 = type(exp1).__name__
```

Comment:
I don't understand what you're trying to test here. Note that you're not testing the case where the setters gets `None`.

Reply:
like this?
```python
    def test_experiment_type(self):
        """Test the experiment_type setter for the experiment."""

        class MyExp(BaseExperiment):
            """Some arbitrary experiment"""

            def __init__(self, *args, **kwargs):
                super().__init__(*args, **kwargs)

            def circuits(self):
                pass

        exp1 = MyExp(physical_qubits=[0], experiment_type="blaaa")
        self.assertEqual(exp1.experiment_type, "blaaa")
        exp2 = MyExp(physical_qubits=[0])
        exp2.experiment_type = "suieee"
        self.assertEqual(exp2.experiment_type, "suieee")
```

Reply:
Yes, and between the third and fourth line add:
```python
self.assertEqual(exp2.experiment_type, "MyExp")
```